components/region_cache_memory_engine/src/engine.rs

```diff
@@ -110,10 +213,12 @@ impl WriteBatchExt for RegionCacheMemoryEngine {
 }
 
 pub struct RegionCacheIterator {
+    cf: String,
     valid: bool,
     prefix_same_as_start: bool,
     prefix: Option<Vec<u8>>,
     iter: IterRef<Skiplist<ByteWiseComparator>, ByteWiseComparator>,
+    // The lower bound is inclusive while the upper bound is exclusive if set
     lower_bound: Option<Vec<u8>>,
```

Comment:
Is it possible to always set lower/uppper bound for this iterator, with region's start and end key if caller has no other bound.  This is the extra safety net to ensure it does not return ranges that does not belong to the region.

Reply:
lower/upper bound is passed through IterOptions where they are Option in it. We can unwrap here to only allow callers with lower/upper bound set to call it.